aggregate_spatial_window.json

```diff
@@ -0,0 +1,95 @@
+{
+  "id": "aggregate_spatial_window",
+  "summary": "Zonal statistics for rectangular windows",
+  "description": "Aggregates statistics over the spatial dimensions. This process passes a list of values to the reducer.\n\nThe results are computed dividing the input X/Y grid into x_size/y_size non-overlapping windows and applying the reducer process to each window.\nIf X and Y are not multiple of x_size/y_size NaNs padding before the computation is applied.",
+  "categories": [
+    "cubes",
+    "aggregate & resample"
+  ],
+  "parameters": [
+    {
+      "name": "data",
+      "description": "A raster data cube.",
+      "schema": {
+        "type": "object",
+        "subtype": "raster-cube"
+      }
+    },
+    {
+      "name": "reducer",
+      "description": "A reducer to be applied on all values of each geometry. A reducer is a single process such as ``mean()`` or a set of processes, which computes a single value for a list of values, see the category 'reducer' for such processes.",
+      "schema": {
+        "type": "object",
+        "subtype": "process-graph",
+        "parameters": [
+          {
+            "name": "data",
+            "description": "An array with elements of any type.",
+            "schema": {
+              "type": "array",
```

Comment:
This passes an flat array, right? So for x_size = 2 and y_size = 2, it could be [5,4,1,7], right? I think this could make the descriptions easier.

Reply:
Well, it depends if the array has the first two dimensions which corresponds to x and y.
If the input has dimension in this order (x,y,band,time) and size [5,4,1,7] and we apply a window with x_size = 2 and y_size = 2 the output would be [2,2,1,7] for boundary 'trim' option or [3,2,1,7] for 'pad'. Is it more clear now?

Reply:
Yeah, maybe we should just take over the definition from https://github.com/Open-EO/openeo-processes/blob/75059c6566a7c8951d04467665252e74d5bb9c3a/apply_neighborhood.json#L45

It has its own subtype `chunk-size` and thus I envision good client support. We can remove the duration of course and may want to remove the null behavior (all values).

Reply:
No, I don't understand this. I assume it's flat, as this is how the spec works.

Reply:
With [5,4,1,7] you mean an array with values 5,4,1 and 7? Which has shape (y,x) = (1,4) ? Then you could use a window with size = [2,2] but you must use _pad_  for applying it.